enterprise/app/workflows/action_list.tsx

```diff
@@ -0,0 +1,212 @@
+import { invocation_status } from "../../../proto/invocation_status_ts_proto";
+import { workflow } from "../../../proto/workflow_ts_proto";
+import format from "../../../app/format/format";
+import { durationToMillis } from "../../../app/util/proto";
+import { GitCommit } from "lucide-react";
+import Link from "../../../app/components/link/link";
+import { MouseCoords, Tooltip, pinBottomLeftOffsetFromMouse } from "../../../app/components/tooltip/tooltip";
+
+export type ActionListComponentProps = {
+  repoUrl: string;
+  history: workflow.ActionHistory[];
+};
+
+type RunStatus = "success" | "failure" | "in-progress" | "disconnected" | "unknown";
+
+function findLatestCompletedRun(
+  actionHistory: workflow.ActionHistory
+): workflow.ActionHistory.ActionHistoryEntry | undefined {
+  return (
+    actionHistory.entries.find((v) => v.status === invocation_status.InvocationStatus.COMPLETE_INVOCATION_STATUS) ??
+    undefined
+  );
+}
+
+function entryToStatus(entry?: workflow.ActionHistory.ActionHistoryEntry): RunStatus {
+  if (!entry) {
+    return "unknown";
+  }
+  if (entry.status === invocation_status.InvocationStatus.COMPLETE_INVOCATION_STATUS) {
+    return entry.success ? "success" : "failure";
+  } else if (entry.status === invocation_status.InvocationStatus.PARTIAL_INVOCATION_STATUS) {
+    return "in-progress";
+  } else if (entry.status === invocation_status.InvocationStatus.DISCONNECTED_INVOCATION_STATUS) {
+    return "disconnected";
+  }
+  return "unknown";
+}
+
+function getCardClass(status: RunStatus): string {
+  if (status === "unknown" || status === "disconnected") {
+    return "neutral";
+  }
+  return status;
+}
+
+function getRunStatusText(status: RunStatus): string {
+  switch (status) {
+    case "disconnected":
+      return "Disconnected";
+    case "failure":
+      return "Failed";
+    case "in-progress":
+      return "In Progress";
+    case "success":
+      return "Succeeded";
+    case "unknown":
+    default:
+      return "Unknown";
+  }
+}
+
+const MINIMUM_BAR_HEIGHT = 12;
+const MAXIMUM_BAR_HEIGHT = 36;
+
+function durationToBarHeight(duration: number, maxDuration: number): string {
+  if (maxDuration === 0) {
+    return MAXIMUM_BAR_HEIGHT + "px";
+  }
+  return Math.max(MINIMUM_BAR_HEIGHT, (duration / maxDuration) * MAXIMUM_BAR_HEIGHT) + "px";
+}
+
+function renderTooltipContent(
+  actionHistory: workflow.ActionHistory,
+  lookup: Map<string, workflow.ActionHistory.ActionHistoryEntry>,
+  c: MouseCoords
+): JSX.Element | null {
+  const elem = document.elementFromPoint(c.clientX, c.clientY);
+  if (!elem) {
+    return null;
+  }
+  const invocationId = elem.getAttribute("data-invocation");
+  if (!invocationId) {
+    return null;
+  }
+  if (invocationId === "no-data") {
+    return <div className="action-history-tooltip">No more runs in the last 7 days.</div>;
+  }
+  const entry = lookup.get(invocationId);
+
+  if (entry) {
+    return (
+      <div className="action-history-tooltip">
```

Comment:
Just double-checking that the styling still looks good for this after nesting under `.workflows-page` - I think the tooltip stuff renders in a portal so it might not be nested under `.workflows-page` anymore. (I usually hack around this by wrapping the tooltip content with a `.workflows-page` div)

Reply:
gah, you're right, fixed.